ask-smapi-sdk/ask_smapi_sdk/smapi_builder.py

```diff
@@ -0,0 +1,195 @@
+    object.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        # type: () -> None
+        """Abstract SmapiClient Builder for building
+        :py:class:`ask_smapi_model.services.skill_management.SkillManagementServiceClient`
+        object.
+        """
+        self._api_endpoint = None  # type: str
+
+    @property
+    def api_endpoint(self):
+        # type: () -> str
+        """Returns the Endpoint to hit by the SMAPI Service.
+
+        :return: Endpoint to hit by the SMAPI service client.
+        :rtype: str
+        """
+        return self._api_endpoint
+
+    @api_endpoint.setter
+    def api_endpoint(self, value):
+        # type: (str) -> None
+        """Sets the Endpoint value to hit by the SMAPI Service.
+
+        :param value: Endpoint to hit by the SMAPI service client.
+        :type value: str
+        """
+        self._api_endpoint = value
+
+    @abstractmethod
+    def client(self):
+        # type: () -> SkillManagementServiceClient
+        raise NotImplementedError
+
+
+class StandardSmapiClientBuilder(SmapiClientBuilder):
+    """Standard SmapiClient Builder class used to generate
+    :py:class:`ask_smapi_model.services.skill_management.SkillManagementServiceClient`
+    object with default Serializer and ApiClient implementations.
+
+    :param client_id: The ClientId value from LWA profiles.
+    :type client_id: str
+    :param client_secret: The ClientSecret value from LWA profiles.
+    :type client_secret: str
+    :param refresh_token: Client refresh_token required to get access token
+        for API calls.
+    :type refresh_token: str
+    """
+
+    def __init__(self, client_id, client_secret, refresh_token):
+        # type: (str, str, str) -> None
+        """Smapi Builder class used to generate
+        :py:class:`ask_smapi_model.services.skill_management.SkillManagementServiceClient`
+        object with default Serializer and ApiClient implementations.
+
+        :param client_id: The ClientId value from LWA profiles.
+        :type client_id: str
+        :param client_secret: The ClientSecret value from LWA profiles.
+        :type client_secret: str
+        :param refresh_token: Client refresh_token required to get access token
+        for API calls.
+        """
+        super(StandardSmapiClientBuilder, self).__init__()
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.refresh_token = refresh_token
+
+    def client(self):
+        # type: () -> SkillManagementServiceClient
+        """Creates the smapi client object using AuthenticationConfiguration
+        and ApiConfiguration registered values.
+
+        :return: A smapi object that can be used for making SMAPI method
+            invocations.
+        :rtype: :py:class:`ask_smapi_model.services.skill_management.SkillManagementServiceClient`
+        """
+        if self._api_endpoint is None:
+            self._api_endpoint = DEFAULT_API_ENDPOINT
```

Comment:
nit: Better to use the `getter` and `setter` than handling the actual attribute here, in case we want to do some processing in either methods.